### ISSUE TITLE
Iterate over list content types (IContentTypes)

### ISSUE THREAD

**Issue by MarkusLanger** [pnp/pnpjs#2897]
### What version of PnPjs library you are using

3.x

### Minor Version Number

21.0

### Target environment

SharePoint Framework

### Additional environment details

SPFx 1.18.2
node 18.17.1

### Question/Request

How can I iterate over all list content types?

I get the content types:
`const ct: IContentTypes | undefined = await (await sp?.web.lists.ensureSitePagesLibrary())?.contentTypes();`

And I also get a response with an array of content types but how can I iterate over the array?
![image](https://github.com/pnp/pnpjs/assets/12131918/16a3556b-1bbd-431b-be09-dad5e33bbd64)

**Comment by MarkusLanger** [comment]
If I use my code in one line as above I don't get the "map" function on the array.
I found out that when I split my code it works fine.
`            const list = await sp?.web.lists.ensureSitePagesLibrary();
            if (list) {
                const ct = await list.contentTypes();
                const ret = ct.map(
                    ({ StringId, Name }): IContentType => ({
                        id: StringId,
                        title: Name,
                    })
                );
                console.log('🚀 ~ fetchContentTypes ~ ret:', ret);
            }
`

**Comment by bcameron1231** [comment]
Glad you were able to come to a resolution! This happens sometimes, but it's not an issue with the library but an issue with IDE's ability to handle intellisense. Sometimes it's able to properly infer the return types, and other times it's not. For example, I had your issue when I first tried to reproduce... and then I closed and re-opened VS Code and it started working. 

![image](https://github.com/pnp/pnpjs/assets/7944457/0b81a18f-1b4e-4c5e-9847-376a56b9eeba)
